repos/spack_repo/builtin/packages/rccl_tests/package.py

```diff
@@ -25,21 +25,17 @@ class RcclTests(MakefilePackage):
 
     variant("mpi", default=True, description="with MPI support")
 
-    depends_on("cxx", type="build")  # generated
+    depends_on("cxx", type="build")
+    requires("%cxx=llvm-amdgpu")
```

Comment:
what about a message like this?

```suggestion
    requires("%cxx=llvm-amdgpu", msg="rccl-tests builds only with llvm-amdgpu")
```